File: luogu/P5876/P5876.py
n = int(input())
s = ['' for i in range(1, 55)]
def check(a, idx):
    for i in range(1, n + 1):
        if i == idx:
            continue
        if s[i][0: len(a)] == a:
            return False
    return True
for i in range(1, n + 1):
    s[i] = str(input())
for i in range(1, n + 1):
    for j in range(1, len(s[i]) + 1):
        sub = s[i][0: j]
        if check(sub, i):
            print(sub)
            break

File: luogu/P5876/P5876.cpp
#include<bits/stdc++.h>
using namespace std;
int n;
string s[51];
bool check(string a,int id)
{
    for(int i=1;i<=n;i++)
    {
        if(i==id)continue;
        if(s[i].substr(0,a.size())==a)return false;
    }
    return true;
}
int main()
{
    scanf("%d",&n);
    for(int i=1;i<=n;i++)cin>>s[i];
    for(int i=1;i<=n;i++)
    {
        for(int j=1;j<=s[i].size();j++)
        {
            string sub=s[i].substr(0,j);
            if(check(sub,i))
            {
                cout<<sub<<endl;
                break;
            }
        }
    }
    return 0;
}